### DISCUSSION TITLE
Highlight some `#if XXX` ... `#endif` to comment

### DISCUSSION THREAD

**Discussion by Freed-Wu** [clangd/clangd#1817] [Q&A] [answered]
Is it possible to highlight the following code to comment?

```sh
#if 0
XXX
#endif
```

Moreover, if `compile_command.json` define `-DFOO`, can we highlight the following code to comment?

```sh
#ifundef FOO
XXX
#endif
```

**Comment by Freed-Wu** [comment]
And the following `XXX` should also be specially highlighted.

```c
int main() {
    return EXIT_SUCCESS;
  XXX
}
```

**Comment by HighCommander4** [comment]
What clangd version are you using, and what client?

**Reply by Freed-Wu** [reply]
clangd 14.0.6 and coc-clangd

**Reply by HighCommander4** [reply, marked as answer]
Clangd 14 produces `comment` semantic tokens for inactive preprocessor branches.

The appearance for a particular kind of semantic token is configured on the client side. I'm not sure how to configure it for coc-clangd, you can ask about this at https://github.com/clangd/coc-clangd/issues. Note that https://github.com/clangd/coc-clangd#configurations suggests that semantic highlighting in general needs to be enabled using `semanticTokens.enable`.
